react-popup/my-app/src/App.tsx

```diff
@@ -0,0 +1,57 @@
+import { useState, useRef } from 'react';
+import './App.css';
+import { Popup } from './Popup';
+
+function App() {
+  // to get the actual button on the dom
+  const btnRef = useRef<HTMLButtonElement>(null);
```

Comment:
You don't ever use this. You should have `ref={btnRef}` on the button on line 19.